ios/browser/api/ads/brave_ads.mm

```diff
@@ -1739,6 +1740,31 @@ - (void)toggleDislikeAd:(NSString*)creativeInstanceId
                        /*intentional*/ base::DoNothing());
 }
 
+- (void)clearData:(void (^)())completion {
+  // Ensure the Brave Ads service is stopped before clearing data.
+  if ([self isServiceRunning]) {
+    return completion();
```

Comment:
Is it valid to call this while its running? If not perhaps we should add a `DCHECK` here

Reply:
Changed if statement to `CHECK(![self isServiceRunning]);`